include/engine/guidance/classification_data.hpp

```diff
@@ -0,0 +1,106 @@
+#ifndef OSRM_GUIDANCE_CLASSIFICATION_DATA_HPP_
+#define OSRM_GUIDANCE_CLASSIFICATION_DATA_HPP_
+
+#include "util/simple_logger.hpp"
+
+#include <string>
+#include <unordered_map>
+
+// Forward Declaration to allow usage of external osmium::Way
+namespace osmium
+{
+class Way;
+}
+
+namespace osrm
+{
+namespace engine
+{
+namespace guidance
+{
+
+enum class FunctionalRoadClass : short
+{
+    UNKNOWN = 0,
+    MOTORWAY,
+    MOTORWAY_LINK,
+    TRUNK,
+    TRUNK_LINK,
+    PRIMARY,
+    PRIMARY_LINK,
+    SECONDARY,
+    SECONDARY_LINK,
+    TERTIARY,
+    TERTIARY_LINK,
+    UNCLASSIFIED,
+    RESIDENTIAL,
+    SERVICE,
+    LIVING_STREET,
+    LOW_PRIORITY_ROAD // a road simply included for connectivity. Should be avoided at all cost
+};
+
+inline FunctionalRoadClass functionalRoadClassFromTag(std::string const &value)
```

Comment:
this translation belongs into `extract/` only. Either move it to lua or put it in `extractor_callbacks.cpp`